### PR TITLE
fix: roles and permissions access (resolves #848)

### LINKED ISSUE #848: Attempt to read property "invitations" on null
**`ErrorException`**

Attempt to read property "invitations" on null

:dart: Illuminate\Foundation\Bootstrap\HandleExceptions::handleError
:page_facing_up: /app/resources/views/settings/roles-and-permissions.blade.php

```php
29   </tr>
30   </thead>
31*  @forelse ($membershipable->invitations as $invitation)
32       <tr>
33           <td id="invitation-{{ $invitation->id }}">{{ $invitation->email }}</td>
```
[Open in Flare](https://flareapp.io/errors/1493239/latest)

### PR DESCRIPTION
**Pull request by chosww**
Add a new policy in UserPolicy to give access to /roles-and-permissions page for users belongs to an organization.

This will resolve the issue #848.

### CODE REVIEW

**Comment by chosww** [comment]
I noticed two test cases are failing from the dev branch, and they are:

Session has unexpected errors: 
  
  [
      "The accessibility_and_inclusion_links.0.url is not a valid URL."
  ]
  Failed asserting that true is false.

  at tests/Feature/RegulatedOrganizationTest.php:111
  
  -----------------------------------------------------
  
  Session has unexpected errors: 
  
  [
      "The team_trainings.0.trainer_url is not a valid URL."
  ]
  Failed asserting that true is false.

  at tests/Feature/ProjectTest.php:306

File: tests/Feature/UserSettingsTest.php [hunk 1]
```diff
@@ -355,7 +355,7 @@
     $response->assertOk();
 });
 
-test('users can invite new members to their organization or regulated organization', function () {
+test('users belong to an organization can invite new members to their organization or regulated organization', function () {
```
Comment (greatislander) [inline tests/Feature/UserSettingsTest.php]:
```suggestion
test('users belonging to an organization or regulated organization can invite new members to their organization or regulated organization', function () {
```

File: tests/Feature/UserSettingsTest.php [hunk 2]
```diff
@@ -355,7 +355,7 @@
     $response->assertOk();
 });
```
Comment (greatislander) [inline tests/Feature/UserSettingsTest.php:377]: Please add tests to "users can edit roles and permissions" test to check that:

- [x] Organization users with an organization can access the page
- [x] Regulated organization users without a regulated organization cannot access the page
- [x] Organization users without an organization cannot access the page

**Review by greatislander — CHANGES_REQUESTED** [review]
This looks good to me, @chosww — please add some tests to cover the new behaviour for the roles and permissions page itself as well as the invitations page and it'll be good to merge.

**Comment by greatislander** [comment]
> I noticed two test cases are failing from the dev branch, and they are:
> 
> Session has unexpected errors:
> 
> [ "The accessibility_and_inclusion_links.0.url is not a valid URL." ] Failed asserting that true is false.
> 
> at tests/Feature/RegulatedOrganizationTest.php:111
> 
> Session has unexpected errors:
> 
> [ "The team_trainings.0.trainer_url is not a valid URL." ] Failed asserting that true is false.
> 
> at tests/Feature/ProjectTest.php:306

@chosww Can you paste in the contents of lines 104-112 of RegulatedOrganizationTest.php and lines 293-307 of ProjectTest.php for me? I think I know what's happening but I'm still puzzled as to why I'm not seeing those failures.

**Comment by chosww** [comment]
> > I noticed two test cases are failing from the dev branch, and they are:
> > Session has unexpected errors:
> > [ "The accessibility_and_inclusion_links.0.url is not a valid URL." ] Failed asserting that true is false.
> > at tests/Feature/RegulatedOrganizationTest.php:111
> > Session has unexpected errors:
> > [ "The team_trainings.0.trainer_url is not a valid URL." ] Failed asserting that true is false.
> > at tests/Feature/ProjectTest.php:306
> 
> @chosww Can you paste in the contents of lines 104-112 of RegulatedOrganizationTest.php and lines 293-307 of ProjectTest.php for me? I think I know what's happening but I'm still puzzled as to why I'm not seeing those failures.

RegulatedOrganizationTest lines 104-112:

```
$response = $this->actingAs($user)->put(localized_route('regulated-organizations.update', $regulatedOrganization), [
    'name' => ['en' => $regulatedOrganization->name],
    'service_areas' => ['NU'],
    'accessibility_and_inclusion_links' => [['title' => 'Accessibility Statement', 'url' => 'https://example.com/accessibility']],
    'social_links' => ['facebook' => ''],
    'publish' => 'Publish',
]);
$response->assertSessionHasNoErrors();
$response->assertRedirect(localized_route('regulated-organizations.show', $regulatedOrganization));
```

ProjectTest line 293-307:
```
$response = $this->actingAs($user)->put(localized_route('projects.update-team', $project), [
'team_count' => '42',
'team_languages' => ['en'],
'team_trainings' => [
    ['name' => 'Example Training', 'date' => '2022-04-01', 'trainer_name' => 'Acme Training Co.', 'trainer_url' => 'example.com'],
],
'contact_person_email' => 'me@here.com',
'contact_person_name' => 'Jonny Appleseed',
'preferred_contact_method' => 'email',
'contact_person_response_time' => ['en' => 'ASAP'],
'save_and_previous' => __('Save and previous'),
]);

$response->assertSessionHasNoErrors();
$response->assertRedirect(localized_route('projects.edit', ['project' => $project, 'step' => 1]));
```

File: tests/Feature/UserSettingsTest.php (hunk 1 above)
Comment (greatislander) [inline tests/Feature/UserSettingsTest.php]:
```suggestion
test('users belonging to an organization or regulated organization can invite new members to their organization or regulated organization', function () {
```

**Review by greatislander — CHANGES_REQUESTED** [review]

**Review by greatislander — APPROVED** [review]

**Pull request merged**